Source/WebKit/UIProcess/mac/WebPageProxyMac.mm

```diff
@@ -973,6 +975,22 @@ static inline bool expectsLegacyImplicitRubberBandControl()
     return bounds;
 }
 
+WebContentMode WebPageProxy::effectiveContentModeAfterAdjustingPolicies(API::WebsitePolicies& policies, const WebCore::ResourceRequest& request)
+{
+    Ref preferences = m_preferences;
+    if (preferences->needsSiteSpecificQuirks()) {
+        if (policies.customUserAgent().isEmpty() && customUserAgent().isEmpty()) {
+            // FIXME (263619): This is done here for adding a UA override to tiktok. Should be in a common location.
+            // needsCustomUserAgentOverride() is currently very generic on purpose.
+            // In the future we want to pass more parameters for targeting specific domains.
+            if (auto customUserAgentForQuirk = Quirks::needsCustomUserAgentOverride(request.url(), m_applicationNameForUserAgent))
+                policies.setCustomUserAgent(*customUserAgentForQuirk);
```

Comment:
```suggestion
                policies.setCustomUserAgent(WTFMove(*customUserAgentForQuirk));
```

Reply:
Oh, good idea!

Reply:
(Totally fine to do this as a followup if we need to change a bunch of other methods downstream to take && as well)